cli/pkg/kctrl/local/min_core_client.go

```diff
@@ -471,5 +487,39 @@ func (*ServiceAccounts) Apply(ctx context.Context, serviceAccount *aplcorev1.Ser
 	return nil, nil
 }
 func (sa *ServiceAccounts) CreateToken(ctx context.Context, serviceAccountName string, tokenRequest *authenticationv1api.TokenRequest, opts metav1.CreateOptions) (*authenticationv1api.TokenRequest, error) {
+	if sa.client == nil {
+		// To handle package release scenarios with DownwardsAPI
+		return &authenticationv1api.TokenRequest{}, nil
+	}
 	return sa.client.CreateToken(ctx, serviceAccountName, tokenRequest, opts)
 }
+
+// Minimum required values to
+type MinDiscoveryClient struct{}
+
+var _ discovery.DiscoveryInterface = &MinDiscoveryClient{}
+
+func (*MinDiscoveryClient) ServerVersion() (*version.Info, error) {
+	return &version.Info{Major: "1", Minor: "30.0-invalid", GitVersion: "1.30.0-invalid"}, nil
```

Comment:
Can this cause a problem if the packages are expecting an older version? If the ytt template have a check or if the helm template only include a certain image if the k8s version is 1.28?

Reply:
It also might be ok if we do not support this use case but it has to be documented somewhere.

Reply:
I think we will be documenting it, the expectation right now is that folks are not swapping images because of versions, they would generally be passing it along as env values if the image is expected to behave differently.
We will definitely call out the values being used in the docs.